src/main/java/org/cactoos/http/io/BoundedByteBuffer.java

```diff
@@ -76,35 +85,39 @@ public void offer(final byte add) {
      * @param bytes The bytes to compare to.
      * @return The value {@code true} if the buffer contains exactly
      *  the {@code bytes}.
-     * @todo #60:30min Figure out a more elegant way to compare
-     *  the ring buffer imitation to another buffer,
-     *  and avoid the for loop hell that is in the following method.
+     * @checkstyle ReturnCountCheck (14 lines)
      */
+    @SuppressWarnings("PMD.OnlyOneReturn")
     public boolean equalTo(final byte[] bytes) {
-        boolean result;
-        if (this.full) {
-            result = bytes.length == this.internal.length;
-            for (int idn = this.idx; idn < this.internal.length; idn += 1) {
-                if (!result) {
-                    break;
-                }
-                result = bytes[idn - this.idx] == this.internal[idn];
-            }
-            for (int idn = 0; idn < this.idx - 1; idn += 1) {
-                if (!result) {
-                    break;
-                }
-                result = bytes[idn + this.idx - 1] == this.internal[idn];
-            }
-        } else {
-            result = bytes.length == this.idx;
-            for (int idn = 0; idn < this.idx - 1; idn += 1) {
-                if (!result) {
-                    break;
-                }
-                result = bytes[idn] == this.internal[idn];
+        if (this.size() != bytes.length) {
+            return false;
+        }
+        int index = this.start;
+        for (final byte current : bytes) {
+            if ((int) current != (int) this.internal[index]) {
+                return false;
```

Comment:
This line `return false` is never tested, add a unit test for this case.

Reply:
@ilyakharlamov Added.